TSPL.docc/LanguageGuide/TheBasics.md

```diff
@@ -86,6 +86,72 @@ because the maximum value never changes.
 The current login attempt counter is declared as a variable,
 because this value must be incremented after each failed login attempt.
 
+If a stored value in your code won't change,
+always declare it as a constant with the `let` keyword.
+Use variables only for storing values that need to be able to change.
+
+When you declare a constant or a variable,
+you can give it a value as part of that declaration,
+like the examples above.
+Alternately,
+you can assign its initial value later in the program,
+as long as it's guaranteed to have a value
+before the first time you read from it.
+
+```swift
+var environment = "development"
+let maximumNumberOfLoginAttempts: Int
+// maximumNumberOfLoginAttempts has no value yet.
+
+if environment == "development" {
+    maximumNumberOfLoginAttempts = 100
+} else {
+    maximumNumberOfLoginAttempts = 10
+}
+// Now maximumNumberOfLoginAttempts has a value, and can be read.
+```
+
+<!--
+  - test: `constantsWithDeferredInitialization`
+
+  ```swifttest
+  -> var environment = "development"
+  -> let maximumNumberOfLoginAttempts: Int
+  -> if environment == "development" {
+         maximumNumberOfLoginAttempts = 100
+     } else {
+         maximumNumberOfLoginAttempts = 10
+     }
+  >> print(maxNumberOfLoginAttempts)
+  << 100
+  ```
+-->
+
+In this example,
+the maximum number of login attempts is constant,
+and its value depends on the environment.
+In the development environment,
+it has a value of 100;
+in any other environment, its value is 10.
+Both branches of the `if` statement
+initialize `maximumNumberOfLoginAttempts` with some value,
+which guarantees that the constant always gets a value.
+
+When you define a constant or variable
+without giving it a value,
+Swift analyzes your code at compile time
+and confirms that all possible paths of execution
+are guaranteed to set a value before reading a value.
+This analysis is called *definitive initialization*.
```

Comment:
@chuckdude Is there a preference from Editorial between calling this "definite initialization" vs "definitive initialization"? The [compiler jargon list](https://github.com/apple/swift/blob/main/docs/Lexicon.md#di-definite-initialization--definitive-initialization) lists both terms. The [blog post on developer.apple.com](https://developer.apple.com/swift/blog/?id=28) uses "definitive" in its link to the Wikipedia article titled "Definite assignment analysis".

I prefer "definite" because it seems more accurate: The compiler allows deferred initialization only when it can prove at compile time that a value is definitely assigned on every code path. There's nothing definitive involved, in the sense of being authoritative or final. The name likely comes from the three states a constant/variable can be in during analysis: definitely assigned (the compiler proved it will have a value set, before being read) definitely unassigned (it proved the opposite) and unknown (it couldn't prove either).

If you agree with "definite", I'll dig into who should review that term from the compiler side and we can add it to a style guide.